.github/workflows/dev.yml

```diff
@@ -0,0 +1,96 @@
+name: Dev
+concurrency: dev
+
+on:
+  push:
+    branches:
+      - master
+      - 'release/*'
+  pull_request:
+    branches:
+      - master
+      - 'release/*'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Inject slug/short variables
+        uses: rlespinasse/github-slug-action@v3.x
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+      - name: Login to Docker Hub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+      - name: Build and Push Temporary Tag
+        uses: docker/build-push-action@v2
+        with:
+          push: true
+          tags: squidex/squidex:tmp-dev-${{ env.GITHUB_SHA_SHORT }}
+          build-args: "SQUIDEX__VERSION=4.0.0-dev-${{ env.GITHUB_SHA_SHORT }}"
+          cache-from: type=registry,ref=squidex/squidex:cache-dev
```

Comment:
The problem with the cache seems to be that it grows forever right now. I had to this:

https://github.com/notifo-io/notifo/blob/main/.github/workflows/deploy.yml#L28

Reply:
According to [cache.md](https://github.com/docker/build-push-action/blob/master/docs/advanced/cache.md), this problem only happens when using actions/cache. This method uses the registry, so there should be no issue.

Reply:
For both approaches it is unclear for me what happens when you run it in the background.

Reply:
It prevents redoing some steps.

Honestly, as the Dockerfile is now, a clean setup would only benefit on Stage 3 on the apt-get run, as any COPY invalidates all upper layer caches. 

Since almost all stages start with a COPY command, the benefit is minimal. Would you like to remove?

Reply:
No, it is beneficial:

1. Because i copy the pacakges.json and csproj first i can very often skip npm install and dotnet restore
2. Because I have 2 separate steps for frontend and backend I can skip the frontend if I only change the backend.

Honestly it does not work reliable, but it should be fine.